src/test/java/com/hermesworld/ais/galapagos/topics/controller/TopicControllerTest.java

```diff
@@ -231,7 +231,7 @@ public void testRemoveTopicProducer_notAuthorizedForTopic_negative() {
         ValidatingTopicService topicService = mock(ValidatingTopicService.class);
 
         TopicController controller = new TopicController(topicService, kafkaClusters, applicationsService,
-                namingService);
+                namingService, userService);
 
         // WHEN I am authorized for the producer, but not the topic owning application
         when(applicationsService.isUserAuthorizedFor("app-9")).thenReturn(true);
```

Comment:
Please add two tests, testing the currentUser.isAdmin check (positive and negative case) when the `skipCompatCheck` flag is active.